### ISSUE TITLE
Allow USER@FQDN format for bootstrap (and chef_client?) commands

### ISSUE THREAD

**Issue by patcon** [higanworks/knife-zero#37]
Knife-solo does this and it's pretty convenient:

- https://matschaffer.github.io/knife-solo/#label-Bootstrap+command
- https://github.com/matschaffer/knife-solo/blob/master/lib/knife-solo/ssh_command.rb#L118-L125

It's a little less elegant here, as chef_client takes a QUERY argument right now. It might make a little more sense if #33 were a thing, and a non-query was assumed to be a key for fqdn.

This would allow the process to go from:

```
knife zero bootstrap myproject-vagrant --ssh-user=vagrant --run-list='role[myproject]'
# creates my user on the machine, as it would on a remote server
zero chef_client name:cmha-local --ssh-user=patcon --sudo
```

to this:

```
knife zero bootstrap vagrant@myproject-vagrant --run-list='role[myproject]'
knife zero chef_client patcon@cmha-local --sudo
```

Which seems a bit cleaner for project setup instructions

**Comment by sawanoboly** [comment]
`--ssh-user` was just inherited(copy) from chef-core. 

If limited to the case of bootstraping, using `user@hostname` format could be nice. I think it also supported by the chef-core.

And I think Chef-Server users(like me) are familiar with `QUERY` than knife-solo usage.

**Comment by patcon** [comment]
From the cli docs, I wasn't aware chef-core supported this, but perhaps it would be best-suited there, so knife-zero would get it for free :)

**Comment by patcon** [comment]
Created a core issue here as well: https://github.com/chef/chef/issues/3527

**Comment by patcon** [comment]
OK, submitted a PR upstream. Thanks!

**Comment by sawanoboly** [comment]
includes chef v12.6.0

https://github.com/chef/chef/blob/master/CHANGELOG.md#1260

**Comment by sawanoboly** [comment]
Released as v1.11.0, Thanks :+1:

**Comment by patcon** [comment]
:)
